### ISSUE TITLE
Use Basic neural network subroutines (BNNS) on iOS

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#3001]
Take advantage of the [BNNS library that Apple announced for iOS](https://developer.apple.com/reference/accelerate/1912851-bnns).

**Comment by aselle** [comment]
Thanks for suggesting this. At this time we probably do not have the bandwidth to look into it immediately, so I am marking it as a contributions welcome.

**Comment by khurram18** [comment]
I want to work and enhance using BNNS on iOS. Can anybody suggest from where to start.

**Comment by damienstanton** [comment]
I'd like to help here. I have a well-defined use case in a personal project. 

When I have time to look into it, I'll track on the `bnns` branch on my fork, [here](https://github.com/damienstanton/tensorflow/tree/bnns).

**Comment by hanleyweng** [comment]
These sample projects might help: 
https://developer.apple.com/library/content/samplecode/MPSCNNHelloWorld/Introduction/Intro.html#//apple_ref/doc/uid/TP40017482
https://developer.apple.com/library/content/samplecode/MetalImageRecognition/Introduction/Intro.html#//apple_ref/doc/uid/TP40017385
https://github.com/shu223/iOS-10-Sampler

For custom weights / own models, perhaps this might help: https://www.bignerdranch.com/blog/use-tensorflow-and-bnns-to-add-machine-learning-to-your-mac-or-ios-app/

**Comment by damienstanton** [comment]
I had to jump away from this for a while due to other obligations  but would love to take a look again. Thanks for the links @hanleyweng

**Comment by paiv** [comment]
I demoed MNIST on BNNS with TensorFlow model here: https://github.com/paiv/mnist-bnns

BNNS is somewhat limited in features. One should make performance comparison before committing.

**Comment by cancan101** [comment]
@paiv Wondering if you saw any issues like this: http://stackoverflow.com/questions/42641668/using-basic-neural-network-subroutines-bnn-accelerate

**Comment by cancan101** [comment]
Core ML might be an interesting abstraction to both this and BNNS: https://github.com/tensorflow/tensorflow/issues/10468.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by bhack** [comment]
Can we close this as we have CORE ML delagate in TF lite?

**Comment by rthadur** [comment]
I guess we can close this request. Thank you
